crates/typst/src/layout/place.rs

```diff
@@ -1,11 +1,23 @@
 use crate::foundations::{elem, scope, Cast, Content, Smart};
 use crate::layout::{Alignment, Em, Length, Rel};
 
-/// Places content at an absolute position.
+/// Places content relatively to its parent container.
 ///
-/// Placed content will not affect the position of other content. Place is
-/// always relative to its parent container and will be in the foreground of all
-/// other content in the container. Page margins will be respected.
+/// Placed content can be either overlaid (the default) or floating. Overlaid
+/// content is aligned with the parent container according to the given
+/// [`alignment`]($place.alignment), and shown over any other content added so
+/// far in the container. Floating content is placed at the top or bottom of
+/// the container, displacing other content down or up respectively. In both
+/// cases, the content position can be adjusted with [`dx`]($place.dx) and
+/// [`dy`]($place.dy) offsets without affecting the layout.
+///
+/// # Effect on the position of other elements
+///
+/// Overlaid elements don't take space in the flow of content, but a `place`
+/// call inserts an invisible block-level element in the flow. This can
+/// affect the layout by breaking the current paragraph. To avoid this,
+/// you can wrap the `place` call in a [`box`]($box) when the call is made
+/// in the middle of a paragraph.
```

Comment:
Would be nice to add an example of this, especially because a normal example follows right below. It could also go under the normal example.

Reply:
Good point. I moved the example before this section (with a new example added to illustrate absolute positioning, and with changes to the original example to make it more complementary). And now this section also has its own example with the `box` trick.

Reply:
I'm not sure whether we should really commend `box(place(..))`. There might be cases where it's the best one can do, but I can't immediately think of such a situation.

Zero-sized boxes don't play well with space trimming in justified paragraphs (if the box ends up at the edge of the line).

Reply:
It's somewhat common though, to want to place something relative to an element inside a paragraph. Could we show another way then? The alternative I can think of is with a label, and a place call  outside of the paragraph, using some calculations with `here()` to get the content in the right position. Basically what `pinit` does. But that's rather involved...

Reply:
What's probably okay is to put a zero sized box immediately after or before a word and joined with a ZWJ (zero width joiner). That way they'll stick together.

Reply:
Makes sense. I've updated the text and example accordingly. Alternatively I can remove the whole advice and example, but I think users will keep trying to place things in paragraphs so we'd need to address the issue one way or another...

Reply:
How about this?
```typ
// An inline-placed element that directly attaches
// to the following word.
#let annotate(..args) = {
  box(place(..args))
  sym.zwj
  h(0pt, weak: true)
}

A placed #annotate(square(size: 1cm), dy: 2pt) 
square in my text.
```

![image](https://github.com/user-attachments/assets/bc665e94-b963-4455-b389-7948c8b9da60)


It's a bit nicer to use via the `h(0pt, weak: true)` trick.

Reply:
Very nice! (also rather dense in tricks: 3 in 3 lines of code :-) ). I've updated the PR to show this example.